### DISCUSSION TITLE
[Feature] Mobile: improvement of selection album to backup

### DISCUSSION THREAD

**Discussion by Solarunit** [immich-app/immich#5148] [Feature Request]
### The feature

**Problem.**

At the moment mobile app searching for the all folders with images:

- it populates a lot (hundreds) of folders you don't need e.g. all folders with music albums containing covers
- app becomes slow to scroll because list is too long
- it hard to select the folders that actually needed to backup
- it generates a very long list of excluded folders, information that not needed

**Suggestion:**

It would be nice to select only particular folders that need to backup manually maybe from internal file manager; don't auto-scan all system for images, similar to Simple Gallery.

### Platform

- [ ] Server
- [ ] Web
- [x] Mobile

**Comment by bo0tzz** [comment]
Addition from #6185:


> I think nearly nobody wants to upload all assets from all folders => do not list the "Recent" (Android)
> pre select the camera as default, most users want to save it and only it
> maybe: list selected folders on top of not selected folders, they could be hard to find, if there are a lot of potential folders
> maybe: show a hierarchical presentation of folders, because some apps create a whole hierarchy of folders containing possible assets, we don't want to backup.
> if folders are showing in a flat list, like now, which is also fine, then there should be an easy way (easy for non nerds, but for normal users, not knowing special Unix command syntax), to exclude folders and their subfolders. A kind of file browser should be used to select folders to be excluded together with their subfolders.
>

**Comment by tomiko23lol** [comment]
- Please implement tree folder structure, listing all available folders is really bad idea. Also If I select parent folder I would expect that every child will be included too. Ideally leverage internal file manager for that.
- Problem > when I close app and open it again, I will click on cloud button and folder selection and upload progress is not there. My settings are not being saved or what?

**Comment by muziqaz** [comment]
So this is the state of the Android app right now:
https://github.com/immich-app/immich/discussions/25125
It seems that app went from one extreme (scanning and adding all the folders with media) to nothing (if the phone is empty).
Since not many of Android users can or want to fiddle with default structure of Android filing system, the potential list of folders to be able to select is relatively predictable. 
Camera (or just DCIM), screenshots, Pictures, WhatsApp (if installed), messenger, Facebook, Instagram (though I don't know if that one saves anything locally).

**Reply by Solarunit** [reply]
we just need the same option like in Fossify Gallery to select folders:

- Manage included folders
- Manage excluded folders

**Reply by AFumi39** [reply]
> we just need the same option like in Fossify Gallery to select folders:
> 
> * Manage included folders
> * Manage excluded folders

I think this is the best solution: a whitelist or blacklist selection, that allow you to backup only the albums you want (whitelist) or backup anything (also the new local albums, without the need to add them one by one every time) except the one you don't want to (blacklist) 👍

**Reply by Solarunit** [reply]
yep!

**Reply by tomiko23lol** [reply]
True, it would help, but only if you give it meaningfull structure, I have multiple folders in my phone containing media where names ate the same. Some of them are just backups, some not. How should user know what they select if only think they see is folder name and that is the same on all of them?

![Screenshot_20260306_101852](https://github.com/user-attachments/assets/d445a8cf-2b04-43f8-8d6f-bb38aac7204b)

**Reply by killeroyo** [reply]
The ability to serach within subfolders is a must-have in order to find all relevant files, e.g.: DCIM (selected) / PHOTOGRAPHY_PRO/BURST/DSC_.../....JPG

**Comment by killeroyo** [comment]
New plan, is it possible to implement?

### Experimental “Folder rules” for Android backup: add DCIM/** recursive include (Android 10+)
Problem

Immich mobile backup selection is currently album-centric (based on photo_manager AssetPathEntity / albumId). This means selecting the “DCIM” album does not reliably include assets created under future subfolders such as DCIM/Burst/..., because those often appear as separate albums/buckets. Users want a way to include a folder root such that future subfolders are automatically covered, without manually selecting each new subfolder/album.
Goal (MVP)

Introduce an experimental “Folder rules” section in Backup settings (Android only for now) with a single rule:

    Include DCIM/** recursively (Android 10+ only)
    Impacts actual backup uploads only (do not update UI counters/preview yet)

This should ensure that assets in DCIM/ and any current/future subfolders (e.g. Burst folders created later) are included automatically.
Non-goals (MVP)

    No iOS support
    No Android <10 support
    No exclude rules / exceptions
    Do not integrate with album selection list or UI asset counters yet

Proposed UX

In the backup album selection screen (prefer the drift-based flow), add a new section above albums:

Folder rules (experimental)

    Toggle: “Backup DCIM (recursive)”
    Subtitle: “Android 10+ • includes future subfolders (e.g. Burst)”

Technical approach

    Add a persisted setting (Store/AppSettings) for the toggle:

    experimentalFolderRulesDcimRecursiveEnabled (bool)

    Add a new Android-only MediaStore-backed source to fetch assets by folder prefix:

    Query MediaStore (images + videos) using RELATIVE_PATH LIKE 'DCIM/%' (Android 10+)
    Map results into the existing Asset / BackupCandidate pipeline

    Hook into backup candidate building:

    In BackupService.buildUploadCandidates(...) / _fetchAssetsAndUpdateLastBackup(...), if toggle enabled and Platform.isAndroid:
        Fetch DCIM candidates via the new MediaStore repository and union them with candidates from selected albums.
        Keep existing exclude-album logic unchanged for MVP.

Canonical code locations (for implementation)

UI (recommended drift-based):

    mobile/lib/pages/backup/drift_backup_album_selection.page.dart
    mobile/lib/widgets/backup/drift_album_info_list_tile.dart
    mobile/lib/providers/backup/backup_album.provider.dart

Backup pipeline:

    mobile/lib/services/backup.service.dart

Current album asset retrieval (albumId-based; not suitable for folder prefix):

    mobile/lib/repositories/album_media.repository.dart

Acceptance criteria

    On Android 10+ with the toggle enabled, assets stored under DCIM/ and any nested subfolders are included in backup uploads without manually selecting each subfolder.
    Toggle disabled: behavior unchanged.
    No new permissions beyond existing media read permissions.
    Clear TODOs captured for future work.

TODO / follow-ups

    Add UI indication and asset counters for folder rules.
    Add exclude prefixes (e.g. DCIM/.thumbnails, DCIM/Screenshots).
    Add incremental scanning / date filtering for performance.
    Add Android <10 fallback or explicit “unsupported” messaging.
    Design iOS equivalent (PhotoKit) or keep Android-only.
    Consider converging album selection + folder rules into a unified “backup rules” model persisted in DB (with migration).

Help wanted

This is a cross-cutting change (UI + backup pipeline + Android MediaStore querying). Assistance is welcome, especially for:

    implementing the MediaStore query for RELATIVE_PATH LIKE 'DCIM/%' (Android 10+),
    mapping MediaStore results into existing Asset/BackupCandidate structures,
    validating behavior across different Android vendors/devices and scoped-storage edge cases.

**Comment by TorstenBrocc** [comment]
+1 on making the backup album selection easier to navigate.

There's already a search box (magnifier) at the top, which helps — but it
means typing the name every time, and it doesn't help when you just want to
scan the list for the album you most recently created.

Concrete ask: let us **sort the backup album selection list by creation date
(newest first)**, or expose a sort control on this screen (the way the Albums
view already has sort options). Right now the list has a fixed order that
isn't newest-first, so a freshly created album is buried and you have to
scroll or remember its exact name to search.

Use case: many device albums, but I usually only sync one specific, recently
created album. Newest-first sorting would make it a one-tap find.

Server 3.1.0, iOS app 3.1.
